app/src/components/modal/modal_lock_yo_tokens/index.tsx

```diff
@@ -0,0 +1,425 @@
+`
+
+const ModalLockTokens = (props: Props) => {
+  const {
+    context,
+    fetchBalances,
+
+    isOpen,
+    omenBalance,
+    onClose,
+    setIsModalLockTokensOpen,
+    theme,
+  } = props
+  const { account, cpk, library: provider, networkId } = context
+
+  const omen = new OmenGuildService(provider, networkId)
+
+  const [isLockAmountOpen, setIsLockAmountOpen] = useState<boolean>(false)
+  const [displayLockAmount, setDisplayLockAmount] = useState<BigNumber>(Zero)
+  const [amountToDisplay, setAmountToDisplay] = useState<string>('')
+  const [totalLocked, setTotalLocked] = useState<BigNumber>(Zero)
+  const [userLocked, setUserLocked] = useState<BigNumber>(Zero)
+  const [timestamp, setTimestamp] = useState<number>(0)
+  const [unlockTimeLeft, setUnlockTimeLeft] = useState<number>(0)
+  const [isTransactionModalOpen, setIsTransactionModalOpen] = useState<boolean>(false)
+  const [transactionMessage, setTransactionMessage] = useState<string>('')
+  const [txHash, setTxHash] = useState<any>('')
+  const [txState, setTxState] = useState<TransactionStep>(TransactionStep.idle)
+  const [txNetId, setTxNetId] = useState()
+  const [omenAllowance, setOmenAllowance] = useState<BigNumber>(Zero)
+  const [allowanceState, setAllowanceState] = useState<ButtonStates>(ButtonStates.idle)
+
+  const isApproveVisible =
+    (omenAllowance.isZero() && isLockAmountOpen) ||
+    (allowanceState === ButtonStates.finished && !omenAllowance.isZero())
+
+  useEffect(() => {
+    setAllowanceState(ButtonStates.idle)
+    ;(async () => {
+      await getTokenLockInfo()
+      await fetchAllowance()
+    })()
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [networkId, account, cpk])
+
+  const getTokenLockInfo = async () => {
+    try {
+      if (cpk?.address) {
+        let address
+        if (context.networkId === networkIds.MAINNET && !context.relay) {
+          address = await omen.tokenVault()
+        } else {
+          address = cpk.address
+        }
+
+        const locked = await omen.tokensLocked(address)
+        const total = await omen.totalLocked()
+
+        const nowInSeconds: number = Math.floor(Date.now() / 1000)
+
+        if (locked.timestamp.toNumber() === 0) setUnlockTimeLeft(0)
+        else setUnlockTimeLeft(locked.timestamp.toNumber() - nowInSeconds)
+
+        setTotalLocked(total)
+        setUserLocked(locked.amount)
+        setTimestamp(locked.timestamp.toNumber())
+      }
+    } catch (e) {
+      console.error(`Error while trying to fetch locked token info:  `, e.message)
+    }
+  }
+  const fetchAllowance = async () => {
+    try {
+      if (cpk) {
+        let allowanceAddress
+        if (context.networkId === networkIds.MAINNET && !context.relay) {
+          allowanceAddress = await omen.tokenVault()
+        } else {
+          allowanceAddress = cpk.address
+        }
+
+        const omenAddress = await omen.omenTokenAddress()
```

Comment:
Would be good to grab via `networks.getContractAddress()` for consistency

Reply:
Same with other places

Reply:
aha okay! My thinking was to maybe simplify it like this since it will always fetch appropriate omenTokenAddress for the current guild we are on....And to handle network checking in the OmenGuildService

Reply:
I fixed it so that It's fetched from KnownTokens but it might be worth considering fetching them from directly from Guild

Reply:
I just noticed that `getContractsAddress()` doesn't work in this case so I think your method is better